File: models/shopping-list-item.js
module.exports = (sequelize, DataTypes) => {
  const ShoppingListItem = sequelize.define('ShoppingListItem', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    listId: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    itemId: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    amount: {
      type: DataTypes.DECIMAL(10, 3),
      allowNull: false,
      defaultValue: 1
    },
    unit: {
      type: DataTypes.ENUM('kg', 'lbs', 'pcs', 'g', 'L', 'ml'),
      allowNull: false,
      defaultValue: 'pcs'
    },
    extraData: {
      type: DataTypes.STRING(255),
      allowNull: true,
      defaultValue: null
    },
    sortOrder: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0
    }
  }, {
    timestamps: true,
    tableName: 'shopping_list_items'
  });

  return ShoppingListItem;
};

File: household/lib/models/shopping_list_item.dart
import 'shopping_item.dart';

class ShoppingListItem {
  final int id;
  final int listId;
  final int itemId;
  final ShoppingItem? item;
  final double amount;
  final String unit;
  final String? extraData;
  final int sortOrder;

  ShoppingListItem({
    required this.id,
    required this.listId,
    required this.itemId,
    this.item,
    required this.amount,
    required this.unit,
    this.extraData,
    required this.sortOrder,
  });

  factory ShoppingListItem.fromJson(Map<String, dynamic> json) {
    return ShoppingListItem(
      id: json['id'] as int,
      listId: json['listId'] as int,
      itemId: json['itemId'] as int,
      item: json['item'] != null
          ? ShoppingItem.fromJson(json['item'] as Map<String, dynamic>)
          : null,
      amount: (json['amount'] as num?)?.toDouble() ?? 1.0,
      unit: json['unit'] as String? ?? 'pcs',
      extraData: json['extraData'] as String?,
      sortOrder: json['sortOrder'] as int? ?? 0,
    );
  }
}
